ykcapi/src/llvmapihelper.rs

```diff
@@ -0,0 +1,37 @@
+use crate::sginterp::SGValue;
+use llvm_sys::core::*;
+use llvm_sys::prelude::{LLVMBasicBlockRef, LLVMValueRef};
+use llvm_sys::{LLVMTypeKind};
+
+pub unsafe fn get_basic_block(func: LLVMValueRef, bbidx: u32) -> LLVMBasicBlockRef {
+    let mut bb = LLVMGetFirstBasicBlock(func);
+    for _ in 0..bbidx {
+        bb = LLVMGetNextBasicBlock(bb);
+    }
+    bb
+}
+
+pub unsafe fn get_instruction(bb: LLVMBasicBlockRef, instridx: u32) -> LLVMValueRef {
+    let mut instr = LLVMGetFirstInstruction(bb);
+    for _ in 0..instridx {
+        instr = LLVMGetNextInstruction(instr);
+    }
+    instr
+}
+
+pub unsafe fn parse_const(c: LLVMValueRef) -> SGValue {
```

Comment:
This function doesn't look like it does parsing?

Reply:
Depends on your definition of parsing I guess. ;-) It turns something like `Constant(Type(width: 32), 1` into `u32 1`. I'm open to suggestions.

Reply:
Let's keep "parsing" for "text to something" conversions :p

I guess this function might be called something like `llvm_const_to_sgconst`?

Reply:
Like it.